src/deploy/executor.rs

```diff
@@ -32,7 +22,7 @@ pub struct Executor<F> {
 
 impl<F> Executor<F>
 where
-	F: Fn(&Path, &Path) -> bool,
+	F: Fn(&Path, &Path) -> Result<bool>,
```

Comment:
I dont know if we should allow the user ask function to fail.
I think it would be better to handle errors in the callback itself and then either return true/false.

Reply:
I dont think so